### ISSUE TITLE
pytorch_lightning version issue

### ISSUE THREAD

**Issue by kaitrepte** [QuantumLab-ZY/HamGNN#104]
Hi,

upon training, I see the following error:

module 'pytorch_lightning.utilities.seed' has no attribute 'seed_everything'.

I am using the most up-to-date version (2.6.5).
Please advise whether a different version is needed, or whether the source code can be updated wrt the newest version.

**Comment by QuantumLab-ZY** [comment]
Hi, thanks for reporting this.

The error comes from an API change in newer `pytorch_lightning` releases. In the current codebase, `HamGNN` calls `pl.utilities.seed.seed_everything(...)`, which works with older Lightning versions but is no longer valid in `pytorch_lightning 2.6.5`.

At the moment, the provided environment file is still pinned to `pytorch-lightning=1.5.10`, so the current release is effectively aligned with the 1.x API. That is why you see this issue when using the latest Lightning version.

For now, you have two options:

1. Use the environment/version set recommended by the project, especially `pytorch-lightning 1.5.10`.
2. Update the source to the newer API (`pl.seed_everything(...)`) so that it works with newer Lightning releases.

We will look into updating the code to improve compatibility with recent `pytorch_lightning` versions. Thanks again for pointing this out.

**Comment by kaitrepte** [comment]
I saw a commit saying that compatibility was established for Pytorch 2.x 
Is this the current version available on github?

**Comment by QuantumLab-ZY** [comment]
Thanks for the follow-up.

You are right that the repository already includes some PyTorch / PyTorch Lightning 2.x compatibility fixes, but the specific seeding call reported in this issue was still using the older Lightning API:

```python
pl.utilities.seed.seed_everything(...)
```

This has now been updated to the public Lightning API:

```python
pl.seed_everything(...)
```

pl.seed_everything is supported by both older Lightning 1.x releases and newer Lightning 2.x releases, so this should resolve the error with recent pytorch_lightning versions such as 2.6.5 without requiring a version-specific workaround.
I also verified the change with:

```python
python -m compileall hamgnn/main.py
python -c "import pytorch_lightning as pl; print(pl.seed_everything)"
```

Both checks passed.

**Comment by kaitrepte** [comment]
Hi,

thanks for clarifying.
The major issue I saw after fixing the source code wrt seed_everything was that 
the Trainer class inside pytorch_lightning has changed almost all its inputs.
Have you adjusted those as well?

Best,
Kai

**Comment by QuantumLab-ZY** [comment]
Thanks for the follow-up.

You are right that full compatibility with the Lightning 2.x API would require updating a few more code paths beyond `seed_everything`, especially the `Trainer` arguments and some epoch-end hooks. The required source changes are not very large, and we originally planned to make them immediately.

However, because HamGNN still needs to preserve compatibility with the historical training setup and checkpoints, we are being cautious about changing the training stack too aggressively. For the current release, the recommended solution is to create the Python environment from the provided `HamGNN.yaml` file in the repository root:

```bash
conda env create -f ./HamGNN.yaml
conda activate HamGNN
```

This environment is intended to be fully compatible with the current HamGNN codebase and pins the dependency versions accordingly, including pytorch-lightning=1.5.10. Using this environment should avoid the Lightning 2.x API incompatibilities you encountered.
We will continue evaluating full Lightning 2.x support, but for now please use the provided HamGNN.yaml environment for the most reliable setup.

**Comment by kaitrepte** [comment]
Ok.

Please let me know once the full compatibility is achieved.

Thanks,
Kai

**Comment by kaitrepte** [comment]
Hi,

any updates on this?

Thanks,
Kai

**Comment by kaitrepte** [comment]
Is there a requirements.txt file that one can use instead of the yaml file which requires conda?

**Comment by QuantumLab-ZY** [comment]
Hi Kai,

A PyTorch Lightning 2.x compatible version is available on the `feat/pl2x-delta-learning` branch:

https://github.com/QuantumLab-ZY/HamGNN/tree/feat/pl2x-delta-learning

We understand that a pip-based installation is useful for virtual environments, containers, and systems where Conda is unavailable. However,
simply converting the YAML file to `requirements.txt` would not be sufficient: PyTorch and PyTorch Geometric packages may require different wheels depending on the CUDA version.

For now, please refer to this branch for Lightning 2.x support.

Best,
Yang

**Comment by QuantumLab-ZY** [comment]
Hi Kai,

Thanks for your patience. HamGNN has now been upgraded and fully supports PyTorch Lightning 2.x. The current `main` branch includes the required Lightning 2.x API updates, including the modern `Trainer` configuration, checkpoint resume handling, progress bar callbacks, lifecycle hooks, and the public `pl.seed_everything(...)` API.

The recommended environment is defined in `HamGNN.yaml`, which uses PyTorch 2.5 and Lightning 2.5.x:

```bash
conda env create -f HamGNN.yaml
conda activate HamGNN
```

Please use the current `main` branch rather than the earlier feature branch. The repository is available here:
https://github.com/QuantumLab-ZY/HamGNN

Best,
Yang

**Comment by kaitrepte** [comment]
Thank you.
Let me get this version and try it out.
Thank you for all the effort!
